api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -757,10 +757,11 @@ type AWSNodePoolPlatform struct {
 	// InstanceProfile is the AWS EC2 instance profile, which is a container for an IAM role that the EC2 instance uses.
 	InstanceProfile string `json:"instanceProfile,omitempty"`
 
-	// Subnet is the subnet to use for node instances.
+	// +kubebuilder:validation:XValidation:rule="has(self.id) && self.id.startsWith('subnet-') ? !has(self.filters) : size(self.filters) > 0", message="subnet is invalid, a valid subnet id or filters must be set, but not both"
```

Comment:
I was thinking that this validation would be done on AWSResourceReference. Wherever we use it, I don't think it's ever valid to have no id and no filter, or an id and a filter.

Reply:
I see, yeah I saw it used in other places as well, and though I didn't want to miss with those.

Reply:
note for future reference: this is not possible due to rule budge
```
The CustomResourceDefinition "nodepools.hypershift.openshift.io" is invalid:
 spec.versions[1].schema.openAPIV3Schema.properties[spec].properties[platform].properties[aws].properties[securityGroups].items.x-kubernetes-validations[0].rule: Forbidden: 
 estimated rule cost exceeds budget by factor of 1.048576x (try simplifying the rule, or adding maxItems, maxProperties, and maxLength where arrays, maps, and strings are declared
```